inlang/source-code/templates/plugin/src/plugin.test.ts

```diff
@@ -18,9 +19,12 @@ it("should return fake messages to illustrate how a plugin works", async () => {
 	await fs.mkdir("/project.inlang", { recursive: true })
 	await fs.writeFile("/project.inlang/settings.json", JSON.stringify(settings))
 
+	// opening the repository
+	const repo = await openRepository("file://", { nodeishFs: fs })
+
```

Comment:
@janfjohannes seems like @jldec is doing the same mistake that @LorisSigrist did by using `file://`? 

What's up with the `file://` stuff? Why are people doing this?

Reply:
@samuelstroschein the intent was to use the existing memoryFs in each of the tests, and instead of passing that fs to loadProject, first open a "mock" repo, and pass the repo instead.

I got the pattern from https://github.com/opral/monorepo/blob/main/lix/source-code/client/src/mockRepo.ts#L16 , but looking at `mockRepo()` again, it think I may be missing the `isoGit.init()` call in the code I wrote for this PR. 

I'm confident that @janfjohannes will sort me out :)

Reply:
@jldec  sorry it took me so long, i was packed with work today...

in tests where you know the test repo is at the root of the memory fs, its ok to just open the root of the filesystem with file:// this is only an isssue in real world code where we do not know where the repo lives on the folder hirarchy. its preobably best to talk about this tomorrow!

Reply:
> @janfjohannes its ok to just open the root of the filesystem with file:// this is only an isssue in real world code where we do not know where the repo lives on the folder hirarchy. its preobably best to talk about this tomorrow!

Too complicated. Let's have a sync call tomorrow about `mockRepo` vs `virtualRepo` and untangle this.  

@jldec can you form an opinion about the discussion in https://github.com/opral/monorepo/pull/2020#discussion_r1453536347? 

- we have confusion with mock repo and virtual repo
- both you and @LorisSigrist used `file://` which is not supposed to happen

Reply:
@jldec  the sdk would try to generate a project id based on the initial hash of the first commit , this would be currenlty the only git action that will (silently) fail if you dont add the repo.

Reply:
See https://github.com/opral/monorepo/pull/2060/files#diff-0d60343fd6cc0079f3185c0adb5a4df993c263b76cc9b506db7f3f469dbd3537

we can get rid of `file://` in tests by calling mockRepo first, and then using `repo.nodeishFs`

```typescript
const repo = await mockRepo()
const fs = repo.nodeishFs

await fs.mkdir(...)
await fs.writeFile(...)

const project = await loadProject({
	projectPath: "/yada/yada.inlang",
	repo,
})
```